clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -94,6 +95,91 @@ def SInt16 : SInt<16>;
 def SInt32 : SInt<32>;
 def SInt64 : SInt<64>;
 
+//===----------------------------------------------------------------------===//
+// FloatType
+//===----------------------------------------------------------------------===//
+
+def CIR_FloatType : CIR_Type<"Float", "float",
+    [DeclareTypeInterfaceMethods<DataLayoutTypeInterface>]> {
+  let summary = "CIR floating-point type";
+  let description = [{
+    CIR type that represents C/C++ floating-point types, including `float`, \
+    `double`, and `long double`.
+  }];
+  let parameters = (ins "::mlir::cir::FloatType::Format":$format);
+
+  let builders = [
+    TypeBuilder<(ins CArg<"::mlir::cir::FloatType::Format">:$format)>
+  ];
+
+  let hasCustomAssemblyFormat = 1;
+  let genVerifyDecl = 1;
+
+  let extraClassDeclaration = [{
+    // Floating-point format.
+    enum Format : uint32_t {
+      FF_Binary16,   // IEEE-754 binary16 format
+      FF_Binary32,   // IEEE-754 binary32 format
+      FF_Binary64,   // IEEE-754 binary64 format
+      FF_Binary128,  // IEEE-754 binary128 format
+      FF_X86Fp80,    // x86 80-bit floating point format
+      FF_BFloat16,   // Brain floating format
+      FF_PpcFp128,   // PowerPC 128-bit floating point format
+    };
+
+    /// Return true if values of this floating-point type is in IEEE-754
+    /// binary16 format.
```

Comment:
Remove all non-relevant stuff here and elsewhere.

Reply:
Updated.